### PR TITLE
pythonPackages.ifcopenshell: init at 0.6.0b

### LINKED ISSUE #72468: Packaging requests: IfcOpenShell (python module)
**Project description**
Open source IFC library and geometry engine
base c++
http://ifcopenshell.org/python.html

**Metadata**

* homepage URL: http://ifcopenshell.org/
* source URL: https://github.com/IfcOpenShell/IfcOpenShell
* license: lgpl 3.0
* platforms: linux

### PR DESCRIPTION
**Pull request by fehnomenal**
###### Motivation for this change

ifcopenshell is missing.

Fixes #72468

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fehnomenal** [comment]
FYI: I'm using this package since the creation of this PR with no problems together with FreeCAD and I'd love to get this merged.
Are there any blockers (besides a huge pile of PRs)?

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2817,6 +2817,12 @@
     githubId = 541748;
     name = "Felipe Espinoza";
   };
+  fehnomenal = {
+    email = "fehnomenal@fehn.systems";
+    github = "fehnomenal";
+    githubId = 9959940;
+    name = "Andreas Fehn";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2824]: Please put this into a separate commit with the message: ``maintainers: add fehnomenal``.

File: pkgs/development/python-modules/ifcopenshell/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  nativeBuildInputs = [ gcc10 cmake ];
+
+  buildInputs = [
+    boost172
+    icu
+    pcre
+    libxml2
+  ];
+
+  patches = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ifcopenshell/default.nix:39]: Please move this under src.

**Review by SuperSandro2000 — COMMENTED** [review]
> besides a huge pile of PRs

Nope.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99246` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python27Packages.ifcopenshell</li>
    <li>python37Packages.ifcopenshell</li>
    <li>python38Packages.ifcopenshell</li>
  </ul>
</details>

File: pkgs/development/python-modules/ifcopenshell/default.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  '';
+
+  cmakeFlags = [
+    "-DOCC_INCLUDE_DIR=${opencascade-occt}/include/opencascade"
+    "-DOCC_LIBRARY_DIR=${opencascade-occt}/lib"
+    "-DOPENCOLLADA_INCLUDE_DIR=${opencollada}/include/opencollada"
+    "-DOPENCOLLADA_LIBRARY_DIR=${opencollada}/lib/opencollada"
+    "-DSWIG_EXECUTABLE=${swig}/bin/swig"
+    "-DLIBXML2_INCLUDE_DIR=${libxml2.dev}/include/libxml2"
+    "-DLIBXML2_LIBRARIES=${libxml2.out}/lib/libxml2.so"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ifcopenshell/default.nix:58]: This line makes it fail on darwin:

```
make[2]: *** No rule to make target '/nix/store/02008j3yn1hs92wncj170j3przajm7j9-libxml2-2.9.10/lib/libxml2.so', needed by 'examples/IfcParseExamples'.  Stop.
make[2]: *** Waiting for unfinished jobs....
```

```
 tree /nix/store/02008j3yn1hs92wncj170j3przajm7j9-libxml2-2.9.10/lib/
/nix/store/02008j3yn1hs92wncj170j3przajm7j9-libxml2-2.9.10/lib/
├── libxml2.2.dylib
├── libxml2.dylib -> libxml2.2.dylib
└── libxml2.la

0 directories, 3 files
```

It should probably be ``libxml2.dylib`` on darwin.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99246` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python27Packages.ifcopenshell</li>
    <li>python37Packages.ifcopenshell</li>
    <li>python38Packages.ifcopenshell</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (fehnomenal) [reply maintainers/maintainer-list.nix:2892]: Done.

File: pkgs/development/python-modules/ifcopenshell/default.nix (hunk 2 above)
Comment (fehnomenal) [reply pkgs/development/python-modules/ifcopenshell/default.nix]: I hope you meant below src. If yes, done :)

File: pkgs/development/python-modules/ifcopenshell/default.nix (hunk 3 above)
Comment (fehnomenal) [reply pkgs/development/python-modules/ifcopenshell/default.nix]: I used `if stdenv.isDarwin then "libxml2.dylib" else "libxml2.so"` which I could test under linux and hopefully works with darwin.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99246` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python27Packages.ifcopenshell</li>
    <li>python37Packages.ifcopenshell</li>
    <li>python38Packages.ifcopenshell</li>
  </ul>
</details>
